iso15118/secc/transport/udp_server.py

```diff
@@ -60,10 +61,24 @@ async def __init__.
         # Allows address to be reused
         sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
 
-        # Bind the socket to the predefined port for receiving
-        # UDP packets (SDP requests)
-        full_ipv6_address = await get_link_local_full_addr(SDP_SERVER_PORT, iface)
-        sock.bind(full_ipv6_address)
+        # Bind the socket to the predefined port on specified interface for receiving
+        # UDP packets (SDP requests). This is done differently on Mac and Linux.
+        # Reference:
+        # https://djangocas.dev/blog/linux/linux-SO_BINDTODEVICE-and-mac-IP_BOUND_IF-to-bind-socket-to-a-network-interface/ # noqa
+        # https://linux.die.net/man/7/socket
+        # https://stackoverflow.com/questions/20616029/os-x-equivalent-of-so-bindtodevice # noqa
+        if platform == "darwin":
+            full_ipv6_address = await get_link_local_full_addr(SDP_SERVER_PORT, iface)
+            sock.bind(full_ipv6_address)
+        else:
+            if not hasattr(socket, "SO_BINDTODEVICE"):
+                socket.SO_BINDTODEVICE = 25
```

Comment:
you can import IN:
```python
import IN

sock.setsockopt(socket.SOL_SOCKET, IN.SO_BINDTODEVICE, ...)
```

Reply:
From the tests done on the board today, the hasattr check isn't required either. Have removed the check.